File: spec/java/src/io/kaitai/struct/spec/TestEnumIf.java
package io.kaitai.struct.spec;

import io.kaitai.struct.testformats.EnumIf;
import org.testng.annotations.Test;

import static org.testng.Assert.assertEquals;

public class TestEnumIf extends CommonSpec {
    @Test
    public void testEnumIf() throws Exception {
        EnumIf r = EnumIf.fromFile(SRC_DIR + "if_struct.bin");

        assertEquals(r.op1().opcode(), EnumIf.Opcodes.A_STRING);
        assertEquals(r.op1().argStr().str(), "foo");

        assertEquals(r.op2().opcode(), EnumIf.Opcodes.A_TUPLE);
        assertEquals(r.op2().argTuple().num1(), 0x42);
        assertEquals(r.op2().argTuple().num2(), 0x43);

        assertEquals(r.op3().opcode(), EnumIf.Opcodes.A_STRING);
        assertEquals(r.op3().argStr().str(), "bar");
    }
}

File: spec/cpp_stl/test_enum_if.cpp
#include <boost/test/unit_test.hpp>

#include <enum_if.h>

#include <iostream>
#include <fstream>
#include <vector>

BOOST_AUTO_TEST_CASE(test_enum_if) {
    std::ifstream ifs("src/if_struct.bin", std::ifstream::binary);
    kaitai::kstream ks(&ifs);
    enum_if_t* r = new enum_if_t(&ks);

    BOOST_CHECK_EQUAL(r->op1()->opcode(), enum_if_t::OPCODES_A_STRING);
    BOOST_CHECK_EQUAL(r->op1()->arg_str()->str(), "foo");

    BOOST_CHECK_EQUAL(r->op2()->opcode(), enum_if_t::OPCODES_A_TUPLE);
    BOOST_CHECK_EQUAL(r->op2()->arg_tuple()->num1(), 0x42);
    BOOST_CHECK_EQUAL(r->op2()->arg_tuple()->num2(), 0x43);

    BOOST_CHECK_EQUAL(r->op3()->opcode(), enum_if_t::OPCODES_A_STRING);
    BOOST_CHECK_EQUAL(r->op3()->arg_str()->str(), "bar");

    delete r;
}
